libs/state/src/lib/medical-illustration/medical-illustration.state.ts

```diff
@@ -31,6 +31,24 @@ export class MedicalIllustrationState {
     setState({ url: url, node: undefined });
   }
 
+  /**
+   * Sets uri by finding it from reference organs
+   * @param param0 state context
+   * @param param1 Action object with iri
+   */
+  @Action(SetUriFromIRI)
+  setUriFromIRI({ patchState, getState }: MedicalIllustrationContext, { iri }: SetUriFromIRI) {
+    const referenceOrgans = getState().referenceOrgans;
+    if (referenceOrgans?.length) {
```

Comment:
Use `referenceOrgans?.find(...)` and throw an error if no matching organ is found